lib/services/nativescript-migration-service.ts

```diff
@@ -85,92 +117,185 @@ export class NativeScriptMigrationService implements IFrameworkMigrationService
-		let fileName = this.getFileNameByVersion(version);
-		let remotePathUrl = `${this.remoteNativeScriptResourcesPath}/tns_modules/${language}/${fileName}`;
-		let filePath = path.join(this.tnsModulesDirectoryPath, language, fileName);
-		return this.$resourceDownloader.downloadResourceFromServer(remotePathUrl, filePath);
+	private downloadTnsPackage(language: string, version: string): IFuture<void> {
+		if (language === NativeScriptMigrationService.TYPESCRIPT_ABBREVIATION || semver.lt(version, this.nativeScriptMigrationData.wait().modulesNpmMinimumVersion)) {
+			let fileName = this.getFileNameByVersion(version);
+			let remotePathUrl = `${this.remoteNativeScriptResourcesPath}/${NativeScriptMigrationService.TNS_MODULES}/${language}/${fileName}`;
+			let filePath = path.join(this.tnsModulesDirectoryPath, language, fileName);
+			return this.$resourceDownloader.downloadResourceFromServer(remotePathUrl, filePath);
+		}
+
+		return Future.fromResult();
 	}
 
 	private getFileNameByVersion(version: string): string {
-		return `${version}.zip`;
+		let fileSuffix = semver.gte(version, this.nativeScriptMigrationData.wait().modulesNpmMinimumVersion) ? NativeScriptMigrationService.TNS_CORE_MODULES : "";
+		return `${version}${fileSuffix}.zip`;
 	}
 
-	private getObsoleteVersions(): IFuture<string[]> {
+	private getDeletedVersions(): IFuture<string[]> {
 		return ((): string[] => {
 			let migrationData = this.nativeScriptMigrationData.wait();
-			return _.map(migrationData.obsoleteVersions, obsoleteVersion => obsoleteVersion.version);
+			return _.map(migrationData.deletedVersions, deletedVersion => deletedVersion.version);
 		}).future<string[]>()();
 	}
 
-	/**
-	 * Checks if the provided {N} version is obsolete and fails with correct error message in such case.
-	 * If the version is not marked as obsolete, do nothing.
-	 * @param {string} version The version tha has to be checked.
-	 * @returns {IFuture<void>}
-	 */
-	private checkIsVersionObsolete(version: string): IFuture<void> {
+	private warnIfVersionObsolete(version: string): IFuture<void> {
+		return (() => {
+			if (!_.contains(this.getSupportedVersions().wait(), version) && !this.isDeleted(version).wait()) {
+				this.$logger.warn("Your project's version is deprecated and will become obsolete in a future release. You can still develop and build your project but it is recommended that you commit all changes to version control and migrate to a newer version of NativeScript.");
+			}
+		}).future<void>()();
+	}
+
+	private throwIfVersionDeleted(version: string): IFuture<void> {
 		return (() => {
-			if(_.any(this.getObsoleteVersions().wait(), obsoleteVersion => obsoleteVersion === version)) {
+			if (this.isDeleted(version).wait()) {
 				this.$errors.failWithoutHelp(`Your project targets NativeScript ${version}. This version is obsolete and cannot be migrated from the command line. If you want to migrate your project, create a new project and copy over your code and resources.`);
```

Comment:
This version is obsolete and cannot be migrated from the command line. 

->

...This version is obsolete. You can no longer develop, build or migrate your project from the command line...